instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/context/ContextWithParent.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.context;
+
+import io.opentelemetry.context.Context;
+
+public class ContextWithParent {
```

Comment:
Doesn't this look like we need `Context.getParent()` method? @anuraaga ?

Reply:
Don't think so, there's no concept of parenting in `Context`, that's a tracing concept. e.g., `context.with(animal, "cat").with(car, "nissan")`, the intermediate contexts naturally aren't parents, but we have no way of knowing what the "logical parent" is at the context layer.

It's one reason how I've generally encouraged the word `invocationContext` instead of parent, but haven't gotten that to stick :P We need to keep a reference to the invocation context to remount it in asynchronous callbacks. 

Normally we would not need this sort of class since, e.g., both contexts would just get scoped into a lambda callback for an async handler. It seems here we're using it for `InstrumentationContext` - I suspect the reason we can't store them as two fields is because we only support a max of one fields for any given type. If this reasoning makes sense, we may want to move this class to the javaagent API instead of exposing it to library instrumentation which shouldn't really need it and at a glance it seems like a class specific to our javaagent needs.